File: Python3/958. Check Completeness of a Binary Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isCompleteTree(self, root: Optional[TreeNode]) -> bool:
        if not root:
            return True

        q = deque([root])
        complete = True

        while q:
            root = q.popleft()
            if not root:
                complete = False
            else:
                if not complete:
                    return False
                q.append(root.left)
                q.append(root.right)

        return True

File: Java/958. Check Completeness of a Binary Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isCompleteTree(TreeNode root) {
        if (root == null) {
            return true;
        }

        boolean complete = true;
        Queue<TreeNode> q = new LinkedList();
        q.offer(root);

        while (!q.isEmpty()) {
            root = q.poll();
            if (root == null) {
                complete = false;
            } else {
                if (!complete) {
                    return false;
                }
                q.offer(root.left);
                q.offer(root.right);
            }
        }

        return true;
    }
}